test/test.js

```diff
@@ -760,6 +760,71 @@ exports.constructor = {
         test.strictEqual(cleanup.callCount, 1);
         test.done();
     },
+    'from Readable - custom onFinish handler - default to end on error': function (test) {
+        test.expect(2);
+        var clock = sinon.useFakeTimers();
+        var rs = new Stream.Readable();
+        var firstTime = true;
+
+        rs._read = function (size) {
+            // Infinite stream!
+        };
+
+        var error1 = new Error('error1');
+        var error2 = new Error('error2');
+        var s = _(rs, function (_rs, callback) {
+            setTimeout(function () {
```

Comment:
Yes. The linter doesn't like it when I shadow variables in functions. Both `_rs` and `rs` should reference the same `Readable`.